components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,307 @@
+    }
+    #[test]
+    fn test_unicodeset_all() {
+        let expected = vec![0, (MAX as u32) + 1];
+        assert_eq!(UnicodeSet::all().inv_list, expected);
+    }
+    #[test]
+    fn test_unicodeset_bmp() {
+        let expected = vec![0, BMP_MAX + 1];
+        assert_eq!(UnicodeSet::bmp().inv_list, expected);
+    }
+
+    // UnicodeSet membership functions
+    #[test]
+    fn test_unicodeset_contains() {
+        let ex = vec![2, 5, 10, 15];
+        let check = UnicodeSet::try_from(ex).unwrap();
+        assert!(check.contains(2 as char));
+        assert!(check.contains(4 as char));
+        assert!(check.contains(10 as char));
+        assert!(check.contains(14 as char));
+    }
+    #[test]
+    fn test_unicodeset_contains_false() {
+        let ex = vec![2, 5, 10, 15];
+        let check = UnicodeSet::try_from(ex).unwrap();
+        assert!(!check.contains(1 as char));
+        assert!(!check.contains(5 as char));
+        assert!(!check.contains(9 as char));
+        assert!(!check.contains(15 as char));
+        assert!(!check.contains(16 as char));
+    }
+    #[test]
+    fn test_unicodeset_contains_range() {
+        let ex = vec![65, 70, 75, 85];
+        let check = UnicodeSet::try_from(ex).unwrap();
+        assert!(check.contains_range('A'..='E')); // 65 - 69
+        assert!(check.contains_range('K'..'U')); // 75 - 84
+    }
+    #[test]
+    fn test_unicodeset_contains_range_false() {
+        let ex = vec![65, 70, 75, 85];
+        let check = UnicodeSet::try_from(ex).unwrap();
+        assert!(!check.contains_range('!'..'A')); // 33 - 65
+        assert!(!check.contains_range('F'..'K')); // 70 - 74
+        assert!(!check.contains_range('U'..));
+    }
+    #[test]
+    fn test_unicodeset_contains_range_invalid() {
+        let check = UnicodeSet::all();
+        assert!(!check.contains_range('A'..'!')); // 65 - 33
+        assert!(!check.contains_range('A'..'A'));
+    }
+    #[test]
+    fn test_unicodeset_size() {
+        let ex = vec![2, 5, 10, 15];
+        let check = UnicodeSet::try_from(ex).unwrap();
+        assert_eq!(8, check.size());
+        let check = UnicodeSet::all();
+        let expected = (MAX as u32) + 1;
+        assert_eq!(expected as usize, check.size());
+        let check = UnicodeSet {
+            inv_list: Vec::new(),
+        };
+        assert_eq!(check.size(), 0);
+    }
+    #[test]
+    fn test_unicodeset_is_empty() {
+        let check = UnicodeSet { inv_list: vec![] };
+        assert!(check.is_empty());
+        let check = UnicodeSet { inv_list: vec![0] };
+        assert!(check.is_empty());
+    }
+    #[test]
+    fn test_unicodeset_is_not_empty() {
+        let check = UnicodeSet::all();
+        assert!(!check.is_empty());
+    }
+    #[test]
+    fn test_unicodeset_ranges() {
+        let ex = vec![65, 70, 75, 85];
+        let check = UnicodeSet::try_from(ex).unwrap();
+        let mut iter = check.ranges();
+        assert_eq!(iter.next().unwrap(), &65);
```

Comment:
Optional: I find it nicer to read `assert_eq!(iter.next(), Some(65))`

Reply:
agreed and changed